avocado/core/dispatcher.py

```diff
@@ -51,10 +51,18 @@ def plugin_type(self):
         else:
             return self.namespace
 
+    def fully_qualified_name(self, extension):
```

Comment:
Actually thinking about it a bit more, how about supporting those directly from each extension? The extension would be responsible to provide `type` (without the namespace prefix, that's the only little issue) and `fqn`.